libs/wire-subsystems/src/Wire/UserSearchSubsystem/Interpreter.hs

```diff
@@ -0,0 +1,198 @@
+    Member GalleyAPIAccess r,
+    Member IndexedUserStore r
+  ) =>
+  (ES.ExternalDocVersion -> ES.VersionControl) ->
+  Sem r ()
+syncAllUsersWithVersion mkVersion =
+  runConduit $
+    paginateWithStateC getIndexUsersPaginated
+      .| logPage
+      .| mkUserDocs
+      .| Conduit.mapM_ IndexedUserStore.bulkUpsert
+  where
+    logPage :: ConduitT [IndexUser] [IndexUser] (Sem r) ()
+    logPage = Conduit.iterM $ \page -> do
+      info $
+        Log.field "size" (length page)
+          . Log.msg (Log.val "Reindex: processing C* page")
+
+    mkUserDocs :: ConduitT [IndexUser] [(ES.DocId, UserDoc, ES.VersionControl)] (Sem r) ()
+    mkUserDocs = Conduit.mapM $ \page -> do
+      visMap <- fmap Map.fromList . unsafePooledForConcurrentlyN 16 (Set.fromList $ mapMaybe (.teamId) page) $ \t ->
+        (t,) <$> teamSearchVisibilityInbound t
+      let vis indexUser = fromMaybe defaultSearchVisibilityInbound $ flip Map.lookup visMap =<< indexUser.teamId
+          mkUserDoc indexUser = indexUserToDoc (vis indexUser) indexUser
+          mkDocVersion = mkVersion . ES.ExternalDocVersion . docVersion . indexUserToVersion
+      pure $ map (\u -> (docId u.userId, mkUserDoc u, mkDocVersion u)) page
+
+updateTeamSearchVisibilityInboundImpl :: (Member IndexedUserStore r) => TeamStatus SearchVisibilityInboundConfig -> Sem r ()
+updateTeamSearchVisibilityInboundImpl teamStatus =
+  IndexedUserStore.updateTeamSearchVisibilityInbound teamStatus.team $
+    searchVisibilityInboundFromFeatureStatus teamStatus.status
+
+searchUserImpl :: Local UserId -> Text -> Maybe Domain -> Maybe (Range 1 500 Int32) -> Sem r [Contact]
+searchUserImpl = undefined
+
+browseTeamImpl :: UserId -> BrowseTeamFilters -> Maybe (Range 1 500 Int32) -> Maybe PagingState -> Sem r [TeamContact]
+browseTeamImpl = undefined
+
+migrateDataImpl ::
+  ( Member IndexedUserStore r,
+    Member (Error MigrationException) r,
+    Member IndexedUserMigrationStore r,
+    Member UserStore r,
+    Member (Concurrency Unsafe) r,
+    Member GalleyAPIAccess r,
+    Member TinyLog r
+  ) =>
+  Sem r ()
+migrateDataImpl = do
+  unlessM IndexedUserStore.doesIndexExist $
+    throw TargetIndexAbsent
+  IndexedUserStore.ensureMigrationIndex
+  foundVersion <- IndexedUserStore.getLatestMigrationVersion
+  if expectedMigrationVersion > foundVersion
+    then do
+      Log.info $
+        Log.msg (Log.val "Migration necessary.")
+          . Log.field "expectedVersion" expectedMigrationVersion
+          . Log.field "foundVersion" foundVersion
+      forceSyncAllUsersImpl
+      IndexedUserStore.persistMigrationVersion expectedMigrationVersion
+    else do
+      Log.info $
+        Log.msg (Log.val "No migration necessary.")
+          . Log.field "expectedVersion" expectedMigrationVersion
+          . Log.field "foundVersion" foundVersion
+
+-- | Increase this number any time you want to force reindexing.
+expectedMigrationVersion :: MigrationVersion
+expectedMigrationVersion = MigrationVersion 6
```

Comment:
This doesn't seem like the most obvious place for this info. Someone could forget to update this. We should also update docs if they say where this number exists in the code.